ludwig/benchmarking/summary_dataclasses.py

```diff
@@ -66,20 +86,200 @@ class ExperimentsDiff:
+            if isinstance(diff_percentage, float):
+                diff_percentage = round(metric.diff_percentage, 3)
+            ret.append(
+                spacing_str.format(
+                    output_feature_name,
+                    metric_name,
+                    experiment1_val,
+                    experiment2_val,
+                    diff,
+                    diff_percentage,
+                )
+            )
+        return "\n".join(ret)
+
+
+@dataclass
+class ResourceUsageSummary:
+    """Summary of resource usage metrics from one experiment."""
+
+    # Path containing the JSON that stores resource usage metrics for the experiment.
+    path: str
+
+    # The tag with which the code block/function is labeled.
+    code_block_tag: str
+
+    # Dictionary that maps from metric name to their values.
+    metric_to_values: Dict[str, Union[float, int]]
+
+    # Names of metrics for the output feature.
+    metric_names: set
+
+
+@dataclass
+class ResourceUsageDiff:
+    """Store resource usage diffs for two experiments."""
 
-    # Experiment summary of the experimental experiment.
-    experimental_summary: ExperimentSummary
+    # The tag with which the code block/function is labeled.
+    code_block_tag: str
+
+    # Name of the base experiment (the one we benchmark against).
+    base_experiment_name: str
 
-    # List of metric diffs for a metric of the two experiments.
-    metrics: List[MetricDiff]
+    # Name of the experimental experiment.
+    experimental_experiment_name: str
+
+    # `List[Diff]` containing diffs for metric of the two experiments.
+    metrics: List[Diff]
+
+    def to_csv(self, path):
+        csv_str = "{}, {}, {}, {}, {}, {}\n"
+        file_name = f"report_{self.code_block_tag}_{self.base_experiment_name}_{self.experimental_experiment_name}.csv"
+        full_path = os.path.join(path, file_name)
+        with open(full_path, "w") as f:
+            f.write(
+                csv_str.format(
+                    "Code Block Tag",
+                    "Metric Name",
+                    self.base_experiment_name,
+                    self.experimental_experiment_name,
+                    "Diff",
+                    "Diff Percentage",
+                )
+            )
+            for metric in sorted(self.metrics, key=lambda m: m.name):
+                diff_percentage = metric.diff_percentage
+                if isinstance(metric.diff_percentage, float):
+                    diff_percentage = round(metric.diff_percentage, 3)
+                f.write(
+                    csv_str.format(
+                        self.code_block_tag,
+                        metric.name,
+                        metric.base_value_str,
+                        metric.experimental_value_str,
+                        metric.diff_str,
+                        diff_percentage,
+                    )
+                )
+        print("Exported report to", full_path)
+
+    def __str__(self):
```

Comment:
In my experience, overriding the built-in python `__str__` method for a dataclass is unusual/uncommon. Could we define a separate `to_string()` method and use that instead? It can be a class method.